src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala

```diff
@@ -173,14 +174,17 @@ final case class ClientsApiServiceImpl(
 
   override def getClientUsers(clientId: String)(implicit
     contexts: Seq[(String, String)],
-    toEntityMarshallerUUIDarray: ToEntityMarshaller[Seq[UUID]],
+    toEntityMarshallerUUIDarray: ToEntityMarshaller[Seq[SelfcareUser]],
     toEntityMarshallerProblem: ToEntityMarshaller[Problem]
   ): Route = {
 
-    val result: Future[Seq[UUID]] = for {
-      clientUuid <- clientId.toFutureUUID
-      users      <- authorizationProcessService.getClientUsers(clientUuid)
-    } yield users
+    val result: Future[Seq[SelfcareUser]] = for {
+      clientUuid   <- clientId.toFutureUUID
+      selfcareUuid <- getSelfcareIdFutureUUID(contexts)
+      clientUsers  <- authorizationProcessService.getClientUsers(clientUuid)
+      users        <- Future.traverse(clientUsers)(selfcareV2ClientService.getUserById(selfcareUuid, _))
+      usersApi     <- Future.traverse(users)(_.toApi.toFuture)
```

Comment:
We do not actually need parallelism here and we can avoid to take threads useful for other requests
```suggestion
      usersApi     <- users.traverse(_.toApi).toFuture
```